### PR TITLE
feat: flag inscription burned

### PR DESCRIPTION
**Pull request by yubing744**
## Summary

flag inscription burned

- Closes #issue
https://github.com/rooch-network/rooch/issues/2012

### CODE REVIEW

**Comment by yubing744** [comment]
@jolestar help review.

File: frameworks/bitcoin-move/sources/ord.move [hunk 1]
```diff
@@ -124,6 +125,8 @@ module bitcoin_move::ord {
         pointer: Option<u64>,
         // Reserved for extending the Rune protocol
         rune: Option<u128>,
+
+        burned: bool,
```
Comment (jolestar) [inline frameworks/bitcoin-move/sources/ord.move:129]: 这个加成动态字段？不然会影响整体数据结构。导入数据那边也会有影响？

File: frameworks/bitcoin-move/sources/ord.move [hunk 2]
```diff
@@ -352,6 +356,12 @@ module bitcoin_move::ord {
                 let match_output_index = new_sat_point.output_index;
 
                 let match_output = vector::borrow(outputs, (match_output_index as u64));
+
+                let output_script_buf = types::txout_script_pubkey(match_output);
+                if (script_buf::is_op_return(output_script_buf)) {
+                    inscription.burned = true;
+                };
+
```
Comment (jolestar) [inline frameworks/bitcoin-move/sources/ord.move:364]: burn 掉之后 to_address 应该是默认值了。直接把 object::to_frozen 冻结掉？

File: frameworks/bitcoin-move/sources/ord.move (hunk 1 above)
Comment (yubing744) [reply frameworks/bitcoin-move/sources/ord.move:129]: OK，我改成动态字段

File: frameworks/bitcoin-move/sources/ord.move (hunk 2 above)
Comment (yubing744) [reply frameworks/bitcoin-move/sources/ord.move:364]: OK

**Comment by yubing744** [comment]
@jolestar help review again.

**Review by jolestar — APPROVED** [review]

**Review by jolestar — APPROVED** [review]

**Pull request merged**